plugin/src/main/java/com/denizenscript/denizen/objects/LocationTag.java

```diff
@@ -4079,6 +4079,41 @@ else if (material.hasModernData() && material.getModernData() instanceof org.buk
             return new ElementTag(BukkitScriptEvent.inCheckInternal(attribute.context, "is_in tag", object, attribute.getParam(), "is_in tag", "is_in tag"));
         });
 
+        // <--[tag]
+        // @attribute <LocationTag.last_interacted_slot>
+        // @returns ElementTag(Number)
+        // @description
+        // Returns the last interacted slot of a Chiseled Bookshelf inventory.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "last_interacted_slot", (attribute, object) -> {
+            BlockState state = object.getBlockStateForTag(attribute);
+            if (!(state instanceof ChiseledBookshelf)) {
```

Comment:
Should use `instanceof` pattern matching, that way you can get rid of the entire `state` variable and just `getBlockStateForTag instanceof ChiseledBookshelf chiseledBookshelf`.